### PR TITLE
Master studio rebase lpe

### PR DESCRIPTION
**Pull request by kebeclibre**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/start.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ export async function startWebClient(Webclient) {
         translatableAttributes: ["data-tooltip"],
         translateFn: env._t,
     });
+    odoo.mainOwlApp = app;
```
Comment (Polymorphe57) [inline addons/web/static/src/start.js:46]: still there?

File: addons/web/static/src/views/form/form_compiler.js [hunk 2]
```diff
@@ -224,8 +224,7 @@ export class FormCompiler extends ViewCompiler {
         const displayClasses = sheetNode
             ? `d-flex {{ uiService.size < ${SIZES.XXL} ? "flex-column" : "flex-nowrap h-100" }}`
             : "d-block";
-        const stateClasses =
-            "{{ props.record.isVirtual or props.record.isDirty ? 'o_form_dirty' : 'o_form_saved' }}";
+        const stateClasses = "{{ props.record.isDirty ? 'o_form_dirty' : 'o_form_saved' }}";
```
Comment (aab-odoo) [inline addons/web/static/src/views/form/form_compiler.js:227]: need to check with @mcm-odoo 

it's weird to have a brand new record with "o_form_saved" classname.

Also this needs a test.

Note: maybe the culprit here is the tour helper, not the form

File: addons/web/static/src/views/view_compiler.js [hunk 3]
```diff
@@ -474,3 +470,7 @@ export function useViewCompiler(ViewCompiler, rawArch, templates, params) {
     }
     return { ...compiledTemplates };
 }
+
+registry.category("owl_templates_cleanup").add("view_compiler", () => {
+    templateIds = Object.create(null);
+});
```
[the remark this answers is not in the document]
Comment (kebeclibre) [reply addons/web/static/src/views/view_compiler.js:476]: yep, a "va-vite" feature.
The studio compiler adds some stuff to the template on the same arch basis as the standard FormCompiler

Go on some form view => arch is compiled
Enter studio => arch is still compiled without a way to wipeout that cache , then realize that none of the studio feature is present in studio

**Comment by kebeclibre** [comment]
closed in favor of https://github.com/odoo/odoo/pull/101932

**Pull request closed without merging**